### DISCUSSION TITLE
ERROR: RunSVD - Problems with Matrix/irlba

### DISCUSSION THREAD

**Discussion by kdzimm** [stuart-lab/signac#1566] [Q&A] [answered]
When I run the "RunSVD" in the latest version of Signac I get the following error:

Running SVD
Error in irlba(A = t(x = object), nv = n, work = irlba.work, tol = tol) :
  function 'as_cholmod_sparse' not provided by package 'Matrix'
Calls: RunSVD ... RunSVD -> RunSVD.Assay -> RunSVD -> RunSVD.default -> irlba
Execution halted

Has anyone else seen this error recently? I have poked around a lot and it looks like the latest version of Matrix caused some issues with Seurat, but I am not having problems with Seurat. It is the RunSVD command where I run into issues. I have tried downgrading Matrix, but that causes other issues

**Comment by timoast** [comment, marked as answer]
Try reinstalling some packages: irlba, Matrix, SeuratObject. There have been several issues opened in the Signac and Seurat repos that you can take a look at for more suggestions, you can search your error message to find them

**Reply by kayarimagenomics** [reply]
I am having this issue and I reinstalled all of these dependencies. Tried Seurat 4 vs 5, tried matrix 1.6-,2,3,4,&5. I have not been able to fix this issue.

**Reply by kayarimagenomics** [reply]
It was irlba - updated to the latest version and it worked

**Reply by realmungmung** [reply]
how did you update it to the latest version? I installed irlba but still got the same error

**Reply by timoast** [reply]
Please see https://github.com/stuart-lab/signac/issues/1596
